Sync/Keys.swift

```diff
@@ -0,0 +1,167 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Account
+import Shared
+import FxA
+import SwiftyJSON
+
+/**
+ * Swift can't do functional factories. I would like to have one of the following
+ * approaches be viable:
+ *
+ * 1. Derive the constructor from the consumer of the factory.
+ * 2. Accept a type as input.
+ *
+ * Neither of these are viable, so we instead pass an explicit constructor closure.
+ *
+ * Most of these approaches produce either odd compiler errors, or -- worse --
+ * compile and then yield runtime EXC_BAD_ACCESS (see Radar 20230159).
+ *
+ * For this reason, be careful trying to simplify or improve this code.
```

Comment:
Do we know if there still be dragons?

Reply:
No idea. I simply moved this code wholesale.